src/System.Private.CoreLib/shared/System/IO/Stream.cs

```diff
@@ -189,6 +189,72 @@ private int GetCopyBufferSize()
             return bufferSize;
         }
 
+        public virtual void CopyTo(ReadOnlySpanAction<byte, object?> callback, object? state, int bufferSize)
+        {
+            if (callback == null) throw new NotSupportedException();
+
+            CopyTo(new WriteCallbackStream(callback, state), bufferSize);
+        }
+
+        public virtual Task CopyToAsync(Func<ReadOnlyMemory<byte>, object?, CancellationToken, ValueTask> callback, object? state, int bufferSize, CancellationToken cancellationToken)
+        {
+            if (callback == null) throw new NotSupportedException();
+
+            return CopyToAsync(new WriteCallbackStream(callback, state), bufferSize, cancellationToken);
```

Comment:
@jkotas, how much overhead is there / would the complexity be worth it to employ here checks similar to the ones you enabled a few years back where we're able to query whether the concrete type overrides particular methods?
https://github.com/dotnet/coreclr/blob/7a3f0d8da0112e7b3f9b11695494fd0bba1b4a52/src/vm/comutilnative.cpp#L2237
And with all of our recent changes to move some of the native support around method tables to managed, could this be implemented in managed code?

I ask because this is a case where it's reasonably common for a Stream-derived type to not override CopyToAsync, and if it doesn't, there's a much more efficient implementation that can be done for this overload (not allocating this WriteCallbackStream or invoking through the stream, and just having a read/write loop that calls the supplied delegate directly.

Reply:
It would mean implement more FCalls like the existing ones. It is certainly doable if we care.

> could this be implemented in managed code?

The data structures to answer this query are way too complex. This cannot be reasonably done in managed code today.

Also, what makes this hard to do well is that C# does not have a cheap way to create a method descriptor reference that compiles to ldtoken instruction. Related to https://github.com/dotnet/csharplang/issues/712. It would need to use IL patching (like what we do for Unsafe intrinsics), or reflection lookup by name and signature.